kafka-handler/src/test/org/apache/hadoop/hive/kafka/AvroBytesConverterTest.java

```diff
@@ -0,0 +1,70 @@
+package org.apache.hadoop.hive.kafka;
+
+import com.google.common.collect.Maps;
+import io.confluent.kafka.schemaregistry.client.MockSchemaRegistryClient;
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
+import org.apache.avro.Schema;
+import org.apache.hadoop.hive.serde2.avro.AvroGenericRecordWritable;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Map;
+
+/**
+ * Created by Milan Baran on 1/29/19 15:03.
+ */
+public class AvroBytesConverterTest {
+
+    private static SimpleRecord simpleRecord1 = SimpleRecord.newBuilder().setId("123").setName("test").build();
+    private static byte[] simpleRecord1AsBytes;
+
+    /**
+     * Emulate confluent avro producer that add 4 magic bits (int) before value bytes. The int represents the schema ID from schema registry.
+     */
+    @BeforeClass
+    public static void setUp() {
+        Map<String,String> config = Maps.newHashMap();
+        config.put("schema.registry.url","http://localhost");
+        KafkaAvroSerializer avroSerializer = new KafkaAvroSerializer(new MockSchemaRegistryClient());
+        avroSerializer.configure(config, false);
```

Comment:
RE: `false` - have Avro keys been tested as well? How do you set the schema for both the key and the value?

(Not sure if this SerDe even reads Kafka message keys...)

Reply:
SerDe reads the key for sure, but TBH never tested the case where the key is an Avro payload as well. Maybe we should add that. 
Currently the key is treated as binary blob, thus you can use any UDF that can reads it as a raw bytes.

Reply:
At my current client, I've ran across a few topics with Avro keys. It's rare (and most often are just Avro `"string"` anyway), but I was mostly curious.

Reply:
Yea, good idea. Right now as @b-slim said keys are treated as bytes. But it this case we need to add a lot of configuration into actual kafka table properties.